src/Microsoft.TestPlatform.ObjectModel/PublicAPI/PublicAPI.Shipped.txt

```diff
@@ -339,14 +343,15 @@ Microsoft.VisualStudio.TestPlatform.ObjectModel.Client.TestRunChangedEventArgs.N
 Microsoft.VisualStudio.TestPlatform.ObjectModel.Client.TestRunChangedEventArgs.TestRunChangedEventArgs(Microsoft.VisualStudio.TestPlatform.ObjectModel.Client.ITestRunStatistics stats, System.Collections.Generic.IEnumerable<Microsoft.VisualStudio.TestPlatform.ObjectModel.TestResult> newTestResults, System.Collections.Generic.IEnumerable<Microsoft.VisualStudio.TestPlatform.ObjectModel.TestCase> activeTests) -> void
 Microsoft.VisualStudio.TestPlatform.ObjectModel.Client.TestRunChangedEventArgs.TestRunStatistics.get -> Microsoft.VisualStudio.TestPlatform.ObjectModel.Client.ITestRunStatistics
 Microsoft.VisualStudio.TestPlatform.ObjectModel.Client.TestRunCompleteEventArgs
+Microsoft.VisualStudio.TestPlatform.ObjectModel.Client.TestRunCompleteEventArgs.TestRunCompleteEventArgs(Microsoft.VisualStudio.TestPlatform.ObjectModel.Client.ITestRunStatistics stats, bool isCanceled, bool isAborted, System.Exception error, System.Collections.ObjectModel.Collection<Microsoft.VisualStudio.TestPlatform.ObjectModel.AttachmentSet> attachmentSets, System.Collections.ObjectModel.Collection<Microsoft.VisualStudio.TestPlatform.ObjectModel.InvokedDataCollector> invokedDataCollectors, System.TimeSpan elapsedTime) -> void
```

Comment:
This adds the invokedDataCollectors but does not keep the previous constructor, which is not a backwards compatible change in public API. Is there a good reason we should do it this way?

Reply:
I did some test with old SDK and this event is created/used inside runner and serialized/deserialized with translator inside TE.
I see few usage inside UTs 
http://index/?query=TestRunCompleteEventArgs&rightProject=VS(Repo)&file=src%5Cvset%5CAgile%5CTestPlatform%5Csuitesrc%5CCore%5CTDD%5CLogging%5CTestRunCompleteEventArgsBehaviors.cs&rightSymbol=ltzdatsjdorb 
https://grep.app/search?q=TestRunCompleteEventArgs&regexp=true&filter[repo][0]=stryker-mutator/stryker-net

Do you see possible runtime issue?

Anyway as like the below one, I'll add overload, I liked the idea to start to narrow down the public api, but if you feel the risk here it's ok to me.

Reply:
Overload added